### PR TITLE
Correct support data for gamepad(dis)connected_event

### PR DESCRIPTION
**Pull request by lukewarlow**
#### Summary

Correct support data for gamepad(dis)connected_event

Mark Safari support as partial due to it missing the window attributes.

Change Firefox support until version 89 to be partial for the same reason.

#### Test results and supporting details

https://bugs.webkit.org/show_bug.cgi?id=223860 - shows the window attributes have only just been implemented in WebKit so can't be in a Safari release.

https://bugzilla.mozilla.org/show_bug.cgi?id=1697093 - shows the window attributes were implemented in Firefox 89.

### CODE REVIEW

File: api/Window.json [hunk 1]
```diff
@@ -2011,20 +2011,36 @@
...
               "version_added": false
             },
             "oculus": "mirror",
             "opera": "mirror",
             "opera_android": "mirror",
             "safari": {
-              "version_added": "10.1"
+              "version_added": "10.1",
+              "partial_implementation": true,
+              "notes": "The event handler property (<code>ongamepadconnected</code>) is not supported; see <a href='https://webkit.org/b/223860'>bug 223860</a>."
```
Comment (lukewarlow) [inline api/Window.json:2043]: Perhaps this should be using the new `impl_url` property instead to link to the bug?

Comment (queengooborg) [reply api/Window.json:2043]: Linking to the bug in the notes makes sense.  `impl_url` is for linking the bug where the feature _is_ implemented, but the bug is related to the notes specifically!

**Review by queengooborg — DISMISSED** [review]
There should be `version_removed` properties added to the statements where the event handlers weren't supported, but otherwise LGTM!

File: api/Window.json [hunk 2]
```diff
@@ -2011,20 +2011,38 @@
...
-            },
+            "firefox": [
+              {
+                "version_added": "89"
+              },
+              {
+                "version_added": "29",
+                "version_removed": "89",
+                "partial_implementation": true,
+                "notes": "The event handler property (<code>ongamepadconnected</code>) is not supported; see <a href='https://bugzil.la/1697093'>bug 1697093</a>."
```
Comment (foolip) [inline api/Window.json:2022]: Suggesting some changes to make this consistent with https://github.com/mdn/browser-compat-data/pull/17061:

```suggestion
                "notes": "The <code>ongamepadconnected</code> event handler property is not supported. See <a href='https://bugzil.la/1697093'>bug 1697093</a>."
```

File: api/Window.json [hunk 3]
```diff
@@ -2011,20 +2011,38 @@
...
+            ],
+            "firefox_android": [
+              {
+                "version_added": "89"
+              },
+              {
+                "version_added": "32",
+                "version_removed": "89",
+                "partial_implementation": true,
+                "notes": "The event handler property (<code>ongamepadconnected</code>) is not supported; see <a href='https://bugzil.la/1697093'>bug 1697093</a>."
```
Comment (foolip) [inline api/Window.json:2033]:
```suggestion
                "notes": "The <code>ongamepadconnected</code> event handler property is not supported. See <a href='https://bugzil.la/1697093'>bug 1697093</a>."
```

File: api/Window.json (hunk 1 above)
Comment (foolip) [reply api/Window.json:2045]:
```suggestion
              "notes": "The <code>ongamepadconnected</code> event handler property is not supported. See <a href='https://webkit.org/b/223860'>bug 223860</a>."
```

File: api/Window.json [hunk 4]
```diff
@@ -2060,20 +2078,38 @@
...
-            },
+            "firefox": [
+              {
+                "version_added": "89"
+              },
+              {
+                "version_added": "29",
+                "version_removed": "89",
+                "partial_implementation": true,
+                "notes": "The event handler property (<code>ongamepaddisconnected</code>) is not supported; see <a href='https://bugzil.la/1697093'>bug 1697093</a>."
```
Comment (foolip) [inline api/Window.json:2089]:
```suggestion
                "notes": "The <code>ongamepaddisconnected</code> event handler property is not supported. See <a href='https://bugzil.la/1697093'>bug 1697093</a>."
```

File: api/Window.json [hunk 5]
```diff
@@ -2060,20 +2078,38 @@
...
+            ],
+            "firefox_android": [
+              {
+                "version_added": "89"
+              },
+              {
+                "version_added": "32",
+                "version_removed": "89",
+                "partial_implementation": true,
+                "notes": "The event handler property (<code>ongamepaddisconnected</code>) is not supported; see <a href='https://bugzil.la/1697093'>bug 1697093</a>."
```
Comment (foolip) [inline api/Window.json:2100]:
```suggestion
                "notes": "The <code>ongamepaddisconnected</code> event handler property is not supported. See <a href='https://bugzil.la/1697093'>bug 1697093</a>."
```

File: api/Window.json [hunk 6]
```diff
@@ -2060,20 +2078,38 @@
...
               "version_added": false
             },
             "oculus": "mirror",
             "opera": "mirror",
             "opera_android": "mirror",
             "safari": {
-              "version_added": "10.1"
+              "version_added": "10.1",
+              "partial_implementation": true,
+              "notes": "The event handler property (<code>ongamepaddisconnected</code>) is not supported; see <a href='https://webkit.org/b/223860'>bug 223860</a>."
```
Comment (foolip) [inline api/Window.json:2112]:
```suggestion
              "notes": "The <code>ongamepaddisconnected</code> event handler property is not supported. See <a href='https://webkit.org/b/223860'>bug 223860</a>."
```

**Review by foolip — CHANGES_REQUESTED** [review]

**Review by foolip — APPROVED** [review]

**Pull request merged**